### ISSUE TITLE
Documentation missing for toDate(locale,format)

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#5351]
When the enhancement to `toDate()` requested in #729 was added in fa072df85c27ce601f867a7f39c5a5d1bf288dec 9 years ago (!), no documentation was included. The online help and the documentation should be updated to reflect the current code.

The test cases show how the list of format strings can be prepended with a string identifying the locale which should be used for the parsing attempts:

https://github.com/OpenRefine/OpenRefine/blob/fa072df85c27ce601f867a7f39c5a5d1bf288dec/main/tests/server/src/com/google/refine/tests/expr/functions/strings/ToFromConversionTests.java#L134-L139

Docs to be updated:

https://github.com/OpenRefine/OpenRefine/blob/d9c235b5b4f763826ab6e4ff1b1d946389286530/main/resources/com/google/refine/grel/FunctionDescription.properties#L123

https://github.com/OpenRefine/OpenRefine/blob/be2d14e72158dfc1c93dce098f97ed59a7eb7dc8/docs/docs/manual/grelfunctions.md#date-functions-date-functions

**Comment by Ayushkkkkkkk** [comment]
Hi I am a beginner and I am really interested in the contribution can you help me to how to start with this?

**Comment by elroykanye** [comment]
hello @Ayushkkkkkkk , you might want to take a look at https://github.com/OpenRefine/OpenRefine/issues/729 and samples used in the test cases for this function.

**Comment by amyra98** [comment]
Please assign me this issue @elroykanye @tfmorris

**Comment by amyra98** [comment]
I have read all the references given along with it. @tfmorris 
So to confirm I just have to add that now toDate() can parse dates in non-default locale like we can give in the first string can be a locale identifier instead of a format stringand so when a locale identifier is provided, the formatting function will use the formatting rules for that locale, rather than the default formatting rules and it will give us the output date object.
I just have to add this in https://github.com/OpenRefine/OpenRefine/blob/d9c235b5b4f763826ab6e4ff1b1d946389286530/main/resources/com/google/refine/grel/FunctionDescription.properties#L123

**Comment by ostephens** [comment]
A slight aside but that description in `fun_to_date` does not seem correct to me:

>Returns the inputted object converted to a date object. Without arguments, it returns the ISO 8601 extended format. With arguments, you can control the output format. With monthFirst: set false if the date is formatted with the day before the month. With formatN: attempt to parse the date using an ordered list of possible formats. Supply formats based on the SimpleDateFormat syntax: <a href="http://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html">SimpleDateFormat</a>

The arguments available to the `toDate` function control how the inputted object is interpreted as a date, not what the output format is (the output of a successful toDate function will always be a date object) It is also not the case that "Without arguments, it returns the ISO 8601 extended format" - again with or without arguments `toDate` returns a date object

**Comment by tfmorris** [comment]
@amyra98 That's very close. I would slightly tweak your statement to say "the ~~formatting~~ **parsing** function will use the formatting rules for that locale."

Also, as @ostephens pointed out, the current help is completely wrong (looks like it was cut & paste from `toString()` and not updated), so I would suggest starting from scratch and rewriting it.

**Comment by ayushi-8102** [comment]
I would like to work on this issue.
 @amyra98 can we work together on it?

**Comment by amyra98** [comment]
> I would like to work on this issue. @amyra98 can we work together on it?

Sorry, I've already made quite a progress on this issue. I don't think so there's much left to contribute now😅

**Comment by tsukipedia** [comment]
I think this should be closed

**Comment by DiamondMyK** [comment]
Hello, can I still contribute this question? I'm new to open source and want to get involved in some simple tasks to get started. Could you assign this problem to me, please? I would appreciate it!

**Comment by wetneb** [comment]
Hi @DiamondMyK, you have already been assigned #5440. Let's focus on that one and then once you have solved that one, we can assign you other issues. If you are no longer working on #5440, please unassign yourself from it first.

**Comment by DiamondMyK** [comment]
> Hi @DiamondMyK, you have already been assigned #5440. Let's focus on that one and then once you have solved that one, we can assign you other issues. If you are no longer working on #5440, please unassign yourself from it first.

I want to deal with #5690 and #5440 together, they both have something to do with grel. Can you assign it to me? I have a certain understanding of grel today and I am ready to solve the problem. Thank you for your reply!

**Comment by DiamondMyK** [comment]
> No, I'd prefer if you tackle the issues one after the other, in separate pull requests.

Ok, thank you for your reply! I will try to solve the first problem!For the open source community I am still a beginner, thank you for your guidance!

**Comment by Ashwinn11** [comment]
I would like to contribute on this. Can you assign me this issue?

**Comment by AbhishekSharma55** [comment]
Hi there,

I want to contribute to this project and take on this issue as my first task. Could you please assign it to me? I am excited to work towards resolving it and will keep the team updated on my progress.

Thanks

**Comment by wetneb** [comment]
@AbhishekSharma55 you are already assigned to another issue: please tackle this one (or unassign yourself from it) before starting to work on another one.

**Comment by godspeedyou** [comment]
@wetneb As others have mentioned and also as this [one](https://github.com/OpenRefine/OpenRefine/issues/5351#issuecomment-1469370801) states, shouldn't this issue be closed? If not clarification on what else is missing would be of help.

**Comment by wetneb** [comment]
The current in-tool documentation (quoted in the issue description, from [FunctionDescription.properties](https://github.com/OpenRefine/OpenRefine/blob/d9c235b5b4f763826ab6e4ff1b1d946389286530/main/resources/com/google/refine/grel/FunctionDescription.properties#L123)) does not mention the possibility of supplying a locale identifier, nor is it consistent with the rest of the behavior of the function as pointed out by @ostephens above.

However, given the high volume of newcomers attempting and failing to tackle this issue, let's remove the "good first issue" tag, because it seems too complicated to grasp.
